pa.sh

```diff
@@ -1,6 +1,32 @@
 #!/usr/bin/env bash
 
-export PASH_TOP=${PASH_TOP:-${BASH_SOURCE%/*}}
+# Help out with shell configuration
+if [ -z "$PASH_TOP" ]; then
+    me="$(readlink -f "${BASH_SOURCE[0]}")"
+    export PASH_TOP="$(dirname "$me")"
+    export PATH="$PASH_TOP:$PATH"
+    printf '# Add this to your shell configuration.\n\n'
+    printf "  export PASH_TOP='%s'\n" "$PASH_TOP"
+    printf '  export PATH="$PASH_TOP:$PATH"\n\n'
+    printf '# If your shell supports process substitution (bash, zsh, etc.),\n'
+    printf '# then run `. <(%s)` to use this config in your current shell.\n' "$0"
+    exit 0
+fi
+
+# Halt if script was sourced. The above is only a setup aid that
+# should not interfere with the script's normal operation.
+return 2>/dev/null
+
+
+rcfile=~/.pash_init
+if [ ! -f "$rcfile" ]; then
```

Comment:
Do we want to do those everytime? Or maybe just with a pash-init.sh script? I am afraid that they will add to the overhead of the shell script. Also, I don't think we should print to stdout (see 3 lines below).

Reply:
I see two things: what you are saying about script overhead, and what I call _deployment overhead_, such as asking users to perform more steps when installing your software. This is my attempt to do the latter, because I cannot (as a system package) easily reach into an exact user's home directory.

Reply:
I figure it's not unreasonable to make `pa.sh` aid in its own configuration if preconditions aren't set. But I understand if you need to protect other properties of the script. What would you have me do?

Reply:
Specifics for the above: When Pash is installed, the only action the user should need to do is run `pa.sh`. I'm trying to hide `PASH_TOP` and all setup work.

I include with PaSh a system-level `.pash_init` because `~` doesn't really mean anything for users like `_apt`. PaSh crashed when I didn't have this file as a normal user. Command line options okay?

Reply:
EDIT: This code is here because it's not suited for system-level deployment. Is it better to use a command line flag here, or ship a configuration helper?

Reply:
Configuration helper is likely the way to go (for now we can have it in `setup-pash.sh`?), as we're trying to minimize startup overheads.

Reply:
> Configuration helper is likely the way to go (for now we can have it in `setup-pash.sh`?), as we're trying to minimize startup overheads.

`setup-pash.sh` is not suitable as a post-install script. Assuming it was:

```
$ sudo apt install pash # ~ is not correct. Recursively invokes sometimes incorrect package managers.
$ pa.sh                 # Not configured for the user
```

Putting the code in `pa.sh` was meant to avoid adding an additional setup command, but we need the shell configuration stuff to happen on the user's command. I was going for this:

```
$ sudo apt install pash # Installs
$ pa.sh                 # Asks for configuration initially, then helps
```